drivers/lora/lora_basics_modem/lbm_common.c

```diff
@@ -0,0 +1,552 @@
+		params.mod_params.bw = RAL_LORA_BW_250_KHZ;
+		break;
+	case BW_500_KHZ:
+		params.mod_params.bw = RAL_LORA_BW_500_KHZ;
+		break;
+	default:
+		ret = -EINVAL;
+		goto release;
+	}
+
+	/* Store TX parameters for use in the TX functions */
+	data->mod_params = params.mod_params;
+	data->pkt_params = params.pkt_params;
+
+	status = ralf_setup_lora(&config->ralf, &params);
+	ret = status == RAL_STATUS_OK ? 0 : -EIO;
+
+release:
+	modem_release(dev);
+	return ret;
+}
+
+int lbm_lora_send_async(const struct device *dev, uint8_t *msg, uint32_t msg_len,
+			struct k_poll_signal *async)
+{
+	const struct lbm_lora_config_common *config = dev->config;
+	struct lbm_lora_data_common *data = dev->data;
+	ral_status_t status;
+	int ret = 0;
+
+	/* Ensure available, freed by TX done callback */
+	if (!modem_acquire(dev)) {
+		return -EBUSY;
+	}
+
+	/* Configure modem for TX */
+	lbm_driver_antenna_configure(dev, MODE_TX);
+	data->modem_mode = MODE_TX;
+
+	/* Validate that we have a TX configuration */
+	if (data->mod_params.sf == 0) {
+		ret = -EINVAL;
+		goto release;
+	}
+
+	/* Store signal */
+	data->operation_done = async;
+
+	/* Update packet params to override the internal packet length variable.
+	 * This has a huge overhead since it performs many register writes, but is the only
+	 * generic way to update the variable. Why this isn't just done in ral_set_pkt_payload
+	 * is anyones guess.
+	 */
+	data->pkt_params.pld_len_in_bytes = msg_len;
+	status = ral_set_lora_pkt_params(&config->ralf.ral, &data->pkt_params);
+	if (status != RAL_STATUS_OK) {
+		ret = -EINVAL;
+		goto release;
+	}
+
+	/* Set the packet payload */
+	status = ral_set_pkt_payload(&config->ralf.ral, msg, msg_len);
+	if (status != RAL_STATUS_OK) {
+		ret = -EINVAL;
+		goto release;
+	}
+
+	/* Start the transmission */
+	status = ral_set_tx(&config->ralf.ral);
+	if (status != RAL_STATUS_OK) {
+		ret = -EINVAL;
+		goto release;
+	}
+	return 0;
+
+release:
+	modem_release(dev);
+	return ret;
+}
+
+int lbm_lora_send(const struct device *dev, uint8_t *msg, uint32_t msg_len)
+{
+	const struct lbm_lora_config_common *config = dev->config;
+	struct lbm_lora_data_common *data = dev->data;
+	struct k_poll_signal done = K_POLL_SIGNAL_INITIALIZER(done);
+	struct k_poll_event evt =
+		K_POLL_EVENT_INITIALIZER(K_POLL_TYPE_SIGNAL, K_POLL_MODE_NOTIFY_ONLY, &done);
+	uint32_t air_time;
+	int ret;
+
+	/* Trigger the asynchronous send */
+	ret = lbm_lora_send_async(dev, msg, msg_len, &done);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Calculate expected airtime of the packet */
+	air_time = ral_get_lora_time_on_air_in_ms(&config->ralf.ral, &data->pkt_params,
+						  &data->mod_params);
+	LOG_DBG("Expected airtime: %d ms", air_time);
+
+	/* Wait for the packet to finish transmitting.
+	 * Use twice the tx duration to ensure that we are actually detecting
+	 * a failed transmission, and not some minor timing variation between
+	 * modem and driver.
+	 */
+	ret = k_poll(&evt, 1, K_MSEC(2 * air_time));
```

Comment:
I have been testing out this new driver. 
In this line here, I seem to get send failures close to 50% of the time often even though I successfully receive the message on the other end. Line 210 gets triggered as a result and the return from the send function is -11.
If I change  the wait time to 3 * air_time it works well without send errors.

My lora settings are: 
```
    config.frequency = 915100000;
    config.bandwidth = BW_500_KHZ;
    config.datarate = SF_6;
    config.preamble_len = 8;
    config.coding_rate = CR_4_5;
    config.iq_inverted = false;
    config.public_network = false;
    config.tx_power = 14;
```

Reply:
msg_len = 9 air_time = 6.

Reply:
Pushed an update with a 10ms offset. I've left it as 2x since 1.5x requires either a division or float multiplication, neither of which are fast. And this is only the failure path, under normal operation it won't timeout anyway.